### PR TITLE
Improve derived_scalars readme

### PR DESCRIPTION
**Pull request by machina-source**
- Improve markdown spacing/style (was dense, now easier to read)
- Add links for file references where relevant
- Clarify indirect references, all wording kept as-is otherwise
- Update code comment casing for codebase consistency

Also added section headers, horizontal rules, and numbered steps for derived scalar computation.

Open to all feedback, thank you.

---

BEFORE:
DS-README-BEFORE

---

AFTER:
DS-README-AFTER

### CODE REVIEW

File: neuron_explainer/activations/derived_scalars/README.md [hunk 1]
```diff
@@ -1,63 +1,149 @@
...
+
+- Defines a transformation to be applied to an `ActivationsAndMetadata` object such that each layer index of the output is the appropriate layer index for computing a given derived scalar. 
+
+  - For example, sometimes a derived scalar D at layer L (for L in `0...num_layers-1`) is computed by operating on activation A from layer L together with activation B from layer L0 (constant). In this case, we would apply a `ConstantLayerIndexer` to B, such that the activations of the `ActivationsAndMetadata` passed to the scalar deriver are from layer L0 of B, no matter the value of L.
+
+#### `RawActivationStore`
+
+- Contains `ActivationsAndMetadata` stored from a forward and possibly a backward pass, separated by `ActivationLocationType` and `PassType`.
+
+#### `DerivedScalarStore`:
```
Comment (stevenbills) [inline neuron_explainer/activations/derived_scalars/README.md:58]: Drop the trailing colon.

File: neuron_explainer/activations/derived_scalars/README.md [hunk 2]
```diff
@@ -1,63 +1,149 @@
...
-1. Add a new `DerivedScalarType` to the Enum in `scalar_deriver.py`.
+> `ScalarDeriver` is meant to be the primary class used to refer to activations once they are READ from disk. 
+>
+> When WRITTEN to disk, the primary class used is `ActivationLocationType`, since we always want to save the least processed form of the data possible.
+
+---
+
+## How to define a new `DerivedScalarType`
+
+1. Add a new `DerivedScalarType` to the Enum in [scalar_deriver.py](../../../neuron_explainer/activations/derived_scalars/scalar_deriver.py):
```
Comment (stevenbills) [inline neuron_explainer/activations/derived_scalars/README.md:122]: Replace the colon at the end with a period.

**Review by stevenbills — APPROVED** [review]
The new formatting looks great! Thanks for making this change.

File: neuron_explainer/activations/derived_scalars/README.md (hunk 1 above)
Comment (machina-source) [reply neuron_explainer/activations/derived_scalars/README.md]: Fixed this - [Fix derived_scalars readme](https://github.com/openai/transformer-debugger/pull/11/commits/c91b0fd47d813e328832f036f49544db06eb7fac)

File: neuron_explainer/activations/derived_scalars/README.md (hunk 2 above)
Comment (machina-source) [reply neuron_explainer/activations/derived_scalars/README.md]: And fixed this - [Fix derived_scalars readme](https://github.com/openai/transformer-debugger/pull/11/commits/c91b0fd47d813e328832f036f49544db06eb7fac)

**Comment by stevenbills** [comment]
I'm still figuring out the permissions for this repo. I'm curious: do you have the ability to merge this PR now that it's approved? (If so feel free to do so!) Not sure if I need to merge PRs myself.

**Comment by machina-source** [comment]
It does not allow me to merge the PR on my side even though the changes are approved. The UI says write access to the repo is required to merge PRs.

This is what I see:

REQ-WRITE-ACCESS

**Comment by stevenbills** [comment]
Thanks for letting me know! Merging now.

**Pull request merged**